lib/web_ui/lib/src/engine/keyboard_binding.dart

```diff
@@ -273,29 +308,14 @@ class KeyboardConverter {
            (metaDown ? _kDeadKeyMeta : 0);
   }
 
-  // Whether `event.key` should be considered a key name.
+  // Whether `event.key` is a key name, such as "Shift", or otherwise a
+  // character, such as "S" or "ж".
   //
-  // The `event.key` can either be a key name or the printable character. If the
-  // first character is an alphabet, it must be either 'A' to 'Z' ( and return
-  // true), or be a key name (and return false). Otherwise, return true.
+  // A key name always starts with a capitalized character, and has more than
+  // 1 letter.
   static bool _eventKeyIsKeyname(String key) {
     assert(key.isNotEmpty);
-    return isAlphabet(key.codeUnitAt(0)) && key.length > 1;
-  }
-
-  static int _characterToLogicalKey(String key) {
-    // Assume the length being <= 2 to be sufficient in all cases. If not,
-    // extend the algorithm.
-    assert(key.length <= 2);
-    int result = key.codeUnitAt(0) & 0xffff;
-    if (key.length == 2) {
-      result += key.codeUnitAt(1) << 16;
-    }
-    // Convert upper letters to lower letters
-    if (result >= _kCharUpperA && result <= _kCharUpperZ) {
-      result = result + _kCharLowerA - _kCharUpperA;
-    }
-    return result;
+    return isUpperLetter(key.codeUnitAt(0)) && key.length > 1;
```

Comment:
Isn't `key.length > 1` sufficient? Or can non-named keys have more than one character? Would it be true that `key.length > 2` would be enough? Just trying to see if we can avoid checking the code unit range entirely.

Reply:
That's a good point. Although I'm not sure if there is a `KeyboardKey.key` that contains multiple JS characters, all keys that `LocaleMapping` handles are single-character, which is all that this function cares about.

Reply:
I think we can remove this range check for now and revise it when we see a counterexample.

Reply:
Just did a quick check of the VSCode layouts. It turns out that `en-in` has a key `L̥` which is two characters.

Reply:
Do all named keys have a name that is >2 characters?

Reply:
Pretty sure so. Because all named keys are alnum, therefore having only 1 character conflicts with letter keys.

Also this is the standard for `KeyboardEvent.key`: https://w3c.github.io/uievents-key/ , which clearly says that `key` can consist of a letter and a combining code unit (accents).